Eastern-Research-Group/expert-query
# Feature/209 select performance

## Related Issues:
* [EQ-209](https://jira.epa.gov/browse/EQ-209)

## Main Changes:
* Switched to materialized views for columns that require distinct queries. This should run fast more consistentently. I did a test on the dev site and a materialized view for assessmentunitname on the catchment_correspondence table has about 600k rows vs the ~50 million rows of the main table.
  * The `values` service will fall back on a distinct query on the table, if we call this `values` service with filters beyond what any of our materialized views can handle.
  * I made this fairly easy to configure, so we should be able to accomodate dependent filters if EPA wants to. I still think if we go that route that we should keep it very limited.
    * I did go ahead and make some groupings that made sense (i.e., assessmentunitid, assessmentunitname, organizationid, and organizationname in one materialized view).
  * The table config also creates indexes on materialized views in the same way we do for tables. We may be able to remove some of these indexes later on.
* Updated our distinct query to filter out nulls.

## Steps To Test:
1. Ensure your env variables are set to your local db
2. Run the etl
3. Verify it runs successfully and creates materialized views
4. Start the app
5. Verify it runs and the drop down queries still work

## TODO:
- [ ] Merge this in and test on dev site
